### PR TITLE
[TIC-147] Swapped Customer Modals for PopUp and Added Focus Trapping

### PR DESCRIPTION
**Pull request by henrykaus**
### Description
* Swapped modals in adding showing to cart and in the cart with `PopUp`
* Added props to `PopUp` to be able to customize the buttons labels and whether the close button appears
* Added focus trapping to the `PopUp` so that when tabbing through elements, it won't leave the `PopUp`
  * Also added `esc` button functionality to close the `PopUp` if close buttons are provided
* Added hover effects to buttons in cart
* Removed inconsistent dark mode styling
* Renamed some functions and files to be proper casing
* Refactored how the admin dropdown is used to better support tabbing through elements on screen. Previously, the dropdown would be within the button that created it. It is now positioned absolutely near the button that opens and closes it

This is the admin dropdown.
image

### Risks
Shouldn't really be any.

### Validation
E2E tests pass locally in chromium, and visual testing.
image

### Issue
[TIC-147](https://pdx-hkaus.atlassian.net/browse/TIC-147) [TIC-145](https://pdx-hkaus.atlassian.net/browse/TIC-145) #499 

### Operating System
Windows 11

[TIC-147]: https://pdx-hkaus.atlassian.net/browse/TIC-147?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TIC-145]: https://pdx-hkaus.atlassian.net/browse/TIC-145?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: client/src/components/Ticketing/event/EventShowings.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  const handleSubmit = (ticketInfo: any) => {
+    const date = ticketInfo.selectedDate;
+    const dateString = date.toLocaleDateString('en-US', {
+      month: 'short',
+      day: '2-digit',
+    });
+
+    setPopUpMessage(`You added ${ticketInfo.qty} ticket${
+      ticketInfo.qty === 1 ? '' : 's'
+    } to ${title} on ${dateString} to the cart.`);
```
Comment (henrykaus) [inline client/src/components/Ticketing/event/EventShowings.tsx:41]: This was changed

File: client/src/components/Ticketing/event/EventShowings.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+ * Data Handler
+ *
+ * @returns {ReactElement} and returns dispatch(fetchTicketingData())
+ */
+const EventShowings = (): ReactElement => {
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const [show, setShow] = useState(false);
+  const [popUpMessage, setPopUpMessage] = useState('');
```
Comment (henrykaus) [inline client/src/components/Ticketing/event/EventShowings.tsx:26]: This was added

File: client/src/components/Ticketing/event/EventShowings.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        <PopUp
+          title='Success!'
+          message={popUpMessage}
+          primaryLabel='Take me there!'
+          success
+          handleClose={handleClose}
+          handleProceed={navigateToCart}
+          showSecondary={false}
+        />
+      )}
```
Comment (henrykaus) [inline client/src/components/Ticketing/event/EventShowings.tsx:124]: This replaced the old modal

File: client/src/components/Ticketing/cart/Cart.tsx [hunk 4]
```diff
@@ -140,15 +145,15 @@ const Cart = (): ReactElement => {
...
+    openModal();
   };
 
   const navigateToCompleteOrder = () => {
     history('/completeorder');
   };
 
   return (
-    <main className='flex flex-col items-center h-full py-[5rem] px-[1rem] tab:px-[5rem] bg-zinc-200'>
+    <main className='flex flex-col items-center min-h-[calc(100vh-233px)] md:min-h-[calc(100vh-142px)] h-full py-[5rem] px-[1rem] tab:px-[5rem] bg-zinc-200'>
```
Comment (henrykaus) [inline client/src/components/Ticketing/cart/Cart.tsx:156]: I found that on larger displays, i.e 1440p, the footer would be rendered in the center of the screen on the cart, so these classes make the min-height of the cart screen, the screen size minus the height of the footer.

**Review by benfaustin95 — APPROVED** [review]

**Pull request merged**